multi-lingual, multiple encoding documentation

btihen opened the pull request:
Here is a first-draft of handling multiple encodings (often found in multi-lingual environments).

I will be glad to make changes as needed

btihen committed:
multi-lingual, multiple encoding documentation

andrewtimberlake commented on docs/multi_lingual_emails.md:
```diff
@@ -0,0 +1,120 @@
+- `utf-8` (Unicode)
+- And others
+
+When processing these emails, you need to ensure that the character data is correctly converted to a consistent encoding (typically UTF-8) to avoid garbled text or processing errors.
```
Not typically—Elixir is UTF-8 only for strings
Suggestion: When processing emails, you need to ensure that data is converted to UTF-8 for further processing

andrewtimberlake commented on docs/multi_lingual_emails.md:
```diff
@@ -0,0 +1,120 @@
+
+## Configuring Charset Handling in Mailroom
+
+Mailroom supports custom charset handling through the `:parser_opts` configuration. You can provide a `:charset_handler` function that will be called with the charset name and the string to convert.
```
These options are passed directly to the [elixir-mail](https://github.com/DockYard/elixir-mail) library used for parsing.

andrewtimberlake commented on docs/multi_lingual_emails.md:
```diff
@@ -0,0 +1,120 @@
+For unexpected charsets, you have several options:
+
+1. **Pass through the string unchanged** - Works if the string is actually valid UTF-8
+2. **Try a common encoding** - Often `:latin1` can work as a reasonable fallback
```
I’m not sure if this applies universally? Just trying an encoding can produce valid UTF-8 but not a valid representation of the underlying message. Many languages use the character codes above 127 to represent vastly different characters for each language/encoding

andrewtimberlake commented on docs/multi_lingual_emails.md:
```diff
@@ -0,0 +1,120 @@
+3. **Log and discard** - For non-critical applications
+4. **Raise an error** - When correct encoding is essential
+
+## Testing Charset Handling
```
For testing, I would just suggest that library users send messages they expect to process with all expected encodings. How they generate them would be context dependent

andrewtimberlake reviewed (commented):
Thanks, I’ve added some suggestions.

btihen committed:
revisions based on Andrew's feedback

btihen commented:
Hopefully, this addresses most of your feedback and ideas.
I added some extra code ideas for unknown conversions.  
If you find it distracting I can remove it, but thought it might be a reasonable approach for some people.

andrewtimberlake commented:
Closing in favour of #24
